banhmi/src/controllers/auth.ts

```diff
@@ -0,0 +1,99 @@
+import { prisma } from "@/db";
+import { createAccessToken, getHashedPasswordWithPepper, getRandomString, refreshAccessToken, validatePassword } from "@/services/auth";
+import { Request, Response } from "express";
+
+const SALT_LENGTH = 50;
+
+async function registerHandler(_req: Request, res: Response) {
+    // TODO: middleware to check required fields and its type
+    const username = _req.body.username
```

Comment:
Since we use the `_req` param inside the function, there is no need to add the underscore here. I think we should change the param name to `req`.